core/server/api_container/server/api_container_service.go

```diff
@@ -130,6 +137,36 @@ func (apicService ApiContainerService) UploadStarlarkPackage(server kurtosis_cor
 	return nil
 }
 
+func (apicService ApiContainerService) InspectFilesArtifactContents(_ context.Context, args *kurtosis_core_rpc_api_bindings.InspectFilesArtifactContentsRequest) (*kurtosis_core_rpc_api_bindings.InspectFilesArtifactContentsResponse, error) {
+	artifactIdentifier := args.GetFileNamesAndUuid().GetFileUuid()
+	if artifactIdentifier == "" {
+		return nil, stacktrace.NewError("An error occurred because files artifact identifier is empty '%v'", artifactIdentifier)
+	}
+
+	filesArtifact, err := apicService.filesArtifactStore.GetFile(artifactIdentifier)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting files artifact '%v'", artifactIdentifier)
+	}
+
+	fileDescriptions, err := getFileDescriptionsFromArtifact(filesArtifact.GetAbsoluteFilepath())
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting file descriptions from '%v'", artifactIdentifier)
+	}
+	if args.GetFilePath() != "" {
```

Comment:
just to confirm; we first find descriptions for all the files; and then check if it exists for the file in question. We are already fetching all the descriptions, wouldn't it be worth for us to just return all of it back to users? Otherwise they would need to make multiple requests to inspect say more than 3 files?

Reply:
I did this out of simplicity, but you are right. Let's chat offline on the best API for us both (CLI and FE)